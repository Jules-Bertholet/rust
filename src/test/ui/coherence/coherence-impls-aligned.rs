#![feature(negative_impls)]

use std::marker::{Aligned, Copy};

enum TestE {
  A
}

struct MyType;

struct NotSync;
impl !Sync for NotSync {}

impl Aligned for TestE {}
//~^ ERROR E0322

impl Aligned for MyType {}
//~^ ERROR E0322

impl Aligned for (MyType, MyType) {}
//~^ ERROR E0322
//~| ERROR E0117

impl Aligned for &'static NotSync {}
//~^ ERROR E0322

impl Aligned for [MyType] {}
//~^ ERROR E0322
//~| ERROR E0117

impl Aligned for &'static [NotSync] {}
//~^ ERROR E0322
//~| ERROR E0117

impl Aligned for dyn Sync {}
//~^ ERROR E0322
//~| ERROR E0117

fn main() {
}
