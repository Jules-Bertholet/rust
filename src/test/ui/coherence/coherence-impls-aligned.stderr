error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:14:1
   |
LL | impl Aligned for TestE {}
   | ^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:17:1
   |
LL | impl Aligned for MyType {}
   | ^^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:20:1
   |
LL | impl Aligned for (MyType, MyType) {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-aligned.rs:20:1
   |
LL | impl Aligned for (MyType, MyType) {}
   | ^^^^^^^^^^^^^^^^^----------------
   | |                |
   | |                this is not defined in the current crate because tuples are always foreign
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead

error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:24:1
   |
LL | impl Aligned for &'static NotSync {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:27:1
   |
LL | impl Aligned for [MyType] {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-aligned.rs:27:1
   |
LL | impl Aligned for [MyType] {}
   | ^^^^^^^^^^^^^^^^^--------
   | |                |
   | |                this is not defined in the current crate because slices are always foreign
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead

error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:31:1
   |
LL | impl Aligned for &'static [NotSync] {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-aligned.rs:31:1
   |
LL | impl Aligned for &'static [NotSync] {}
   | ^^^^^^^^^^^^^^^^^------------------
   | |                |
   | |                this is not defined in the current crate because slices are always foreign
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead

error[E0322]: explicit impls for the `Aligned` trait are not permitted
  --> $DIR/coherence-impls-aligned.rs:35:1
   |
LL | impl Aligned for dyn Sync {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^ impl of `Aligned` not allowed

error[E0117]: only traits defined in the current crate can be implemented for arbitrary types
  --> $DIR/coherence-impls-aligned.rs:35:1
   |
LL | impl Aligned for dyn Sync {}
   | ^^^^^^^^^^^^^^^^^--------
   | |                |
   | |                `dyn Sync` is not defined in the current crate
   | impl doesn't use only types from inside the current crate
   |
   = note: define and implement a trait or new type instead

error: aborting due to 11 previous errors

Some errors have detailed explanations: E0117, E0322.
For more information about an error, try `rustc --explain E0117`.
